File: backend/internal/models/notification_subscription.go
package models

import (
	"github.com/SherClockHolmes/webpush-go"
)

type Topic string

const (
	RSVP         Topic = "new_rsvp"
	CHAT         Topic = "new_chat"
	GAME_UPDATE  Topic = "game_update"
	EVENT_UPDATE Topic = "event_update"
)

// All the JSON is blanked because this is sensitive information and should never go to the front end
type NotificationSubscription struct {
	DbModel
	Topics []Topic `json:"topics"`
	webpush.Subscription
}

File: app/src/models/NotificationSubscription.ts
import type { DbModel } from './index'

export enum NotifTopic {
  RSVP = 'new_rsvp',
  CHAT = 'new_chat',
  GAME_UPDATE = 'game_update',
  EVENT_UPDATE = 'event_update'
}

export interface NotificationSubscription extends DbModel {
  topics: Array<NotifTopic>
  // TODO: Are we supposed to be getting this webpush stuff?
  endpoint: string
  auth: string
  p265dh: string
}
